### ISSUE TITLE
Scalar Element-wise Power?

### ISSUE THREAD

**Issue by cdeterman** [viennacl/viennacl-dev#155]
I know there is element-wise power with vectors and matrices

e.g.

    viennacl::matrix C = element_pow(viennacl::matrix A, viennacl::matrix B);

Is it possible to use a scalar value instead of allocating an entire matrix of '2's for example?

    viennacl::matrix C = element_pow(viennacl::matrix A, 2);

or

    viennacl::matrix A ^= 2;

**Comment by karlrupp** [comment]
No, this is currently not supported.

**Comment by cdeterman** [comment]
Are any matrix-scalar operations implemented?  If not, I would probably change the title of this issue to just element-wise scalar operations.

**Comment by karlrupp** [comment]
I'm not sure what you're exactly referring to by 'scalar operations'. If you mean

    A *= 2.0;
    A /= 2.0;

then yes, they are available. :-)

**Comment by cdeterman** [comment]
Ah, okay so `*` and `/` are implemented.  But it appears `+`,`-`, and `^` are all not implemented then.  Correct?  I verified that a program compiles with `*=` but errors with `+=`.

**Comment by karlrupp** [comment]
That is correct, yes. Since users have asked for such functionality a few times in the past, it's probably best to provide it. I'll change the title of this issue accordingly.

(I refrain from overloading `^`, though, because it has a different meaning for built-in types.)

**Comment by cdeterman** [comment]
I assume this is still a work in progress.  Any suggested workarounds in the meantime if I want to add a scalar to all matrix elements?  Creating a second matrix of the scalar value can eat up quite a bit of memory in larger instances.  This works fine for multiplication:

```c++
viennacl::matrix<T> vmat;
vmat = vmat * (T)(scalar);
```

but addition is a no go:

```c++
viennacl::matrix<T> vmat;
vmat = vmat + (T)(scalar);
```

**Comment by karlrupp** [comment]
Sorry, I don't have a workaround at the moment other than using `scalar_matrix`. I'll implement it when I find the time for it.
